tsconfig.json

```diff
@@ -1,5 +1,6 @@
 {
-  "include": ["src/**/*", "test/**/*"],
+  "files": ["test/xhr2.d.ts"],
```

Comment:
**suggestion:** I don't think keeping both src config and test config in one `tsconfig.json` should still be our approach here. Editing this tsconfig for the sake of tests might cause similar unwanted behaviour in the future.

I suggest moving this to `tsconfig.test.json`:

```json
{
  "extends": "./tsconfig.json",
  "include": ["test/**/*"]
}
```

Then updating `jest.config.js` to use the new `tsconfig.test.json` file.

Reply:
@TheBuggedYRN That's actually a good suggestion. Let's do it this way. 👍